code/game/machinery/vending/cm_vending.dm

```diff
@@ -565,48 +565,28 @@ GLOBAL_LIST_EMPTY(vending_products)
 								to_chat(user, SPAN_WARNING("Only specialists can take specialist sets."))
 								vend_fail()
 								return FALSE
+
 							else if(!user.skills || user.skills.get_skill_level(SKILL_SPEC_WEAPONS) != SKILL_SPEC_TRAINED)
 								to_chat(user, SPAN_WARNING("You already have a specialization."))
 								vend_fail()
 								return FALSE
+
 							var/p_name = itemspec[1]
-							if(!GLOB.available_specialist_sets.Find(p_name))
+							if(!(p_name in GLOB.specialist_set_name_dict))
+								return
+
+							if(GLOB.specialist_set_name_dict[p_name].get_available_vendor_num() <= 0)
 								to_chat(user, SPAN_WARNING("That set is already taken."))
 								vend_fail()
 								return FALSE
-							var/obj/item/card/id/card = human_user.get_idcard()
-							if(!card?.check_biometrics(user))
+
+							var/obj/item/card/id/card = human_user.wear_id
```

Comment:
```suggestion
							var/obj/item/card/id/card = human_user.get_idcard()
```
Didnt drathek spent bunch of time to move *all* of these to get_idcard?

Reply:
My bad yeah